Yosemite/Yosemite/Stores/Order/OrdersUpsertUseCase.swift

```diff
@@ -304,25 +304,25 @@ struct OrdersUpsertUseCase {
     /// Updates, inserts, or prunes the provided `storageOrder`'s custom fields using the provided `readOnlyOrder`'s custom fields
     ///
     private func handleOrderCustomFields(_ readOnlyOrder: Networking.Order, _ storageOrder: Storage.Order, _ storage: StorageType) {
-        // Upsert the `customFields` from the `readOnlyOrder`
-        readOnlyOrder.customFields.forEach { readOnlyCustomField in
-            if let existingStorageMetaData = storage.loadOrderMetaData(siteID: readOnlyOrder.siteID,
-                                                                       orderID: storageOrder.orderID,
-                                                                       metadataID: readOnlyCustomField.metadataID) {
-                existingStorageMetaData.update(with: readOnlyCustomField)
-            } else {
-                let newStorageMetaData = storage.insertNewObject(ofType: Storage.MetaData.self)
-                newStorageMetaData.update(with: readOnlyCustomField)
-                storageOrder.addToCustomFields(newStorageMetaData)
-            }
-        }
+        // Create a set of metadata IDs from the read-only order for quick lookup
+        let readOnlyMetadataIDs = Set(readOnlyOrder.customFields.map { $0.metadataID })
 
-        // Now, remove any objects that exist in `storageOrder.customFields` but not in `readOnlyOrder.customFields`
+        // Remove any objects that exist in `storageOrder.customFields` but not in `readOnlyOrder.customFields`
```

Comment:
I'll let someone else review this as I also contributed a bit to it, but this comment (and the one for products) should be updated. We are removing all custom fields for the order right?

Reply:
Fixed here 9d34b5c8b8214eb7147ba806b809da1222700c96